### PR TITLE
API CHANGE: honor user-defined precision, rename SamplerPrecision.

### PR DESCRIPTION
**Pull request by prideout**
matc now honors 'precision' on non-samplers in mat files, and
the filamat API now allows clients to specify precision for
non-samplers.

This involved flattening a union that is internal to filamat.

### CODE REVIEW

File: android/filamat-android/src/main/java/com/google/android/filament/filamat/MaterialBuilder.java [hunk 1]
```diff
@@ -259,7 +259,7 @@ public MaterialBuilder uniformParameterArray(@NonNull UniformType type, int size
 
     @NonNull
     public MaterialBuilder samplerParameter(@NonNull SamplerType type, SamplerFormat format,
```
Comment (romainguy) [inline android/filamat-android/src/main/java/com/google/android/filament/filamat/MaterialBuilder.java:261]: You should add new methods for non-sampler parameters to specify the precision. Java doesn't have default parameter values, so just use overloading instead.

**Review by romainguy — CHANGES_REQUESTED** [review]

**Review by pixelflinger — APPROVED** [review]

File: android/filamat-android/src/main/java/com/google/android/filament/filamat/MaterialBuilder.java (hunk 1 above)
Comment (prideout) [reply android/filamat-android/src/main/java/com/google/android/filament/filamat/MaterialBuilder.java:278]: Good catch, added a commit.

**Review by bejado — COMMENTED** [review]
> Will this require a middle version number bump? What about `MATERIAL_VERSION`?

Due to the enum rename, I do think we should bump the middle version number, yes.  It doesn't break backward compatibility of materials so I don't think we need to bump MATERIAL_VERSION.

**Review by romainguy — APPROVED** [review]

**Comment by romainguy** [comment]
Merging this PR because I need it to update mine :)

**Pull request merged**